perun-registrar-lib/src/test/java/cz/metacentrum/perun/registrar/ExpirationNotifSchedulerTest.java

```diff
@@ -232,6 +239,70 @@ public void checkMembersState() throws Exception {
 
 	}
 
+	@Test
+	public void checkVoApplicationShouldBeAutoRejected() throws Exception {
+		System.out.println(CLASS_NAME + "checkVoApplicationShouldBeAutoRejected");
+
+		Application submitApp = setUpAndSubmitAppForPotentialAutoRejection(70, null, VO_APP_EXP_RULES);
+
+		ReflectionTestUtils.invokeMethod(spyScheduler, "voApplicationsAutoRejection", perun.getVosManagerBl().getVos(session));
+
+		// check results
+		Application returnedApp = registrarManager.getApplicationById(session, submitApp.getId());
+		assertEquals("Application should be rejected.", returnedApp.getState(), Application.AppState.REJECTED);
+	}
+
+	@Test
+	public void checkVoApplicationShouldNotBeAutoRejected() throws Exception {
+		System.out.println(CLASS_NAME + "checkVoApplicationShouldNotBeAutoRejected");
+
+		Application submitApp = setUpAndSubmitAppForPotentialAutoRejection(50, null, VO_APP_EXP_RULES);
+
+		ReflectionTestUtils.invokeMethod(spyScheduler, "voApplicationsAutoRejection", perun.getVosManagerBl().getVos(session));
+
+		// check results
+		Application returnedApp = registrarManager.getApplicationById(session, submitApp.getId());
+		assertEquals("Application shouldn't be rejected.", returnedApp.getState(), Application.AppState.VERIFIED);
+	}
+
+	@Test
+	public void checkGroupApplicationShouldBeAutoRejected() throws Exception {
+		System.out.println(CLASS_NAME + "checkGroupApplicationAutoRejection");
+
+		jdbc.update("DELETE from application_form WHERE vo_id = ?", vo.getId());
+
+		Group group = new Group();
+		group.setName("Group for apply");
+		perun.getGroupsManagerBl().createGroup(session, vo, group);
+
+		Application submitApp = setUpAndSubmitAppForPotentialAutoRejection(70, group, GROUP_APP_EXP_RULES);
+
+		ReflectionTestUtils.invokeMethod(spyScheduler, "groupApplicationsAutoRejection", perun.getVosManagerBl().getVos(session));
+
+		// check results
+		Application returnedApp = registrarManager.getApplicationById(session, submitApp.getId());
+		assertEquals("Application should be rejected.", returnedApp.getState(), Application.AppState.REJECTED);
+	}
+	@Test
+	public void checkGroupApplicationShouldNotBeAutoRejected() throws Exception {
+		System.out.println(CLASS_NAME + "checkGroupApplicationAutoRejection");
+
+		jdbc.update("DELETE from application_form WHERE vo_id = ?", vo.getId());
+
+		Group group = new Group();
+		group.setName("Group for apply");
+		perun.getGroupsManagerBl().createGroup(session, vo, group);
+
+		Application submitApp = setUpAndSubmitAppForPotentialAutoRejection(50, group, GROUP_APP_EXP_RULES);
+
+		ReflectionTestUtils.invokeMethod(spyScheduler, "groupApplicationsAutoRejection", perun.getVosManagerBl().getVos(session));
```

Comment:
I would prefer only passing expected VO in the list. Otherwise it would process all VOs in the DB. We sometimes run tests against non-empty devel DB.

Reply:
Changed to singleton list of expected vo